frb_codegen/src/parser/mod.rs

```diff
@@ -65,12 +65,26 @@ impl<'a> Parser<'a> {
         match inner {
             ty::SupportedInnerType::Path(ty::SupportedPathType {
                 ident,
-                generic: Some(generic),
-            }) if ident == RESULT_IDENT => Some(IrFuncOutput::ResultType(
-                self.type_parser.convert_to_ir_type(*generic)?,
-            )),
+                mut generic,
+                is_exception: _,
+            }) if ident == RESULT_IDENT && !generic.is_empty() => {
+                let first_argument = self
+                    .type_parser
+                    .convert_to_ir_type(generic.remove(0), false)
+                    .unwrap();
+                let second_argument = if generic.len() == 1 {
+                    Some(
+                        self.type_parser
+                            .convert_to_ir_type(generic.remove(0), true)
+                            .unwrap(),
+                    )
+                } else {
+                    None
+                };
+                Some(IrFuncOutput::ResultType(first_argument, second_argument))
```

Comment:
uber-nit: improve naming, "first_argument", "second_argument" - maybe it is sth like normal result, error result?